jib-core/src/main/java/com/google/cloud/tools/jib/global/JibSystemProperties.java

```diff
@@ -32,10 +33,12 @@
   public static final String SEND_CREDENTIALS_OVER_HTTP = "sendCredentialsOverHttp";
   public static final String SERIALIZE = "jib.serialize";
 
-  private static final String DISABLE_USER_AGENT = "_JIB_DISABLE_USER_AGENT";
-
   @VisibleForTesting public static final String SKIP_EXISTING_IMAGES = "jib.skipExistingImages";
 
+  @VisibleForTesting
+  static final String USE_WILDCARD_FOR_DEPENDENCY_CLASSPATH =
+      "jib.useWildcardForDependencyClasspath";
```

Comment:
I realize this is a draft, but I'm not super fond of this name. I sometimes think `preserveClasspathOrder` is a decent name, but i'm not sure.

Reply:
Sounds good. And this made me realize that `JibSystemProperties` is for jib-core. The property should be in jib-plugins-common.

Reply:
I think it's natural to negate the name, such as `noClasspathOrderPreserving`. The `PropertyNames` class only defines property names, and `System.getBoolean()` doesn't have a way to return `true` as default (which I think makes sense to avoid confusion).

Reply:
Hrmm.. maybe in that case, your naming makes more sense. `useClasspathWildcard` is similar but shorter?

Also any idea if we would include this as a config option rather than system property?

Reply:
Actually maybe the full name is necessary

Reply:
The idea was to have the flag as a temporary escape hatch that will go away soon once we don't get any report from the user. But I think I was too naive. Previous I speculate that we don't have to worry about the argument length limit (https://github.com/GoogleContainerTools/jib/issues/2733#issuecomment-683015187, https://github.com/GoogleContainerTools/jib/issues/2471#issuecomment-630229008). However, after searching "argument list too long" on the Internet today, I completely lost my previous confidence. Now I think the length limit is real and this change is not safe. For example: https://github.com/gradle/gradle/issues/5754.

So it would be safer to have this as an opt-in feature. But that's really really unfortunate, because issues from a different library order are very cryptic and difficult to identify the cause. I really wish Java 8 supported the `@`-argument file feature.

Anyways, if we do this, now I think we should have it as a config option (and additionally a system property). But the question as to whether we expand it by default or not remains.

But ideally, for Java 9+, I think we should really use the `@`-argument file feature; we can always preserve the classpath order while never hitting the system argument length limit. This is going to be a lot bigger change.

In sum, perhaps
- Java 9+: always use a `@`-argument file. (`preserveClasspathOrder` is irrelevant.)
- Java <=8: expand or not based on `preserveClasspathOrder`. Not decisive on if it should be true by default.

Reply:
I think false by default works (as it is the current behavior) for java8-